src/mustache_core.js

```diff
@@ -470,29 +469,7 @@ var core = {
 	 * @return {String}
 	 */
 	cleanWhitespaceControl: function(template) {
-		return template.replace(mustacheWhitespaceRegExp, function(
-			whole,
-			spaceBefore,
-			bracketBefore,
-			controlBefore,
-			expression,
-			controlAfter,
-			bracketAfter,
-			spaceAfter,
-			matchIndex
-		) {
-
-			if (controlBefore === '-') {
-				spaceBefore = '';
-			}
-
-			if (controlAfter === '-') {
-				spaceAfter = '';
-			}
-
-			return spaceBefore + bracketBefore + expression + bracketAfter + spaceAfter;
-
-		});
+		return template.replace(/\s*\{\{--\}\}\s*|\s*(\{\{\{?)-|-(\}\}\}?)\s*/g, "$1$2");
```

Comment:
generally a good practice to cache regexps.

Reply:
The code in `master` cached it but the regex is using the `g`lobal flag and doesn't reset the `lastIndex` so might be bugged. Doing it inline without cache avoids that hidden bug.

I can cache it and make sure `lastIndex` is reset when I add the doc under `Tags`